src/plugins/intel_cpu/src/nodes/mkldnn_rnn.cpp

```diff
@@ -248,248 +266,207 @@ void MKLDNNRNN::getSupportedDescriptors() {
+        THROW_ERROR << "has incorrect input ranks. Data rank: " << getInputShapeAtPort(0).getRank() <<
+                "; Hidden state rank: " << getInputShapeAtPort(1).getRank();
 
-    cell_type = ie2dnnl(op);
-    cell_act = ie2dnnl(rnnCellBase->get_activations()[0]);  // Works only for RNN with one gate
-
-    auto in_data_dims = op->get_input_shape(0);
-    auto in_h_state_dims = op->get_input_shape(1);
-    auto out_h_state_dims = op->get_output_shape(0);
-
-    if (in_data_dims.size() != 2 || in_h_state_dims.size() != 2)
-        IE_THROW() << "Incorrect shape of input/output ports for layer " << getName();
-
-    G = gatesCount(cell_type);
-    S = statesCount(cell_type);
     T = 1;
-    N  = in_data_dims[0];
-    DC = in_data_dims[1];
-    SC = in_h_state_dims[1];
-
-    Gb = (cell_type != mkldnn::algorithm::lbr_gru) ? G : G + 1;
+    if (cell_type == algorithm::vanilla_lstm)
+        DC = getInputShapeAtPort(3).getDims()[1];
+    else
+        DC = getInputShapeAtPort(2).getDims()[1];
 
-    // Expected shapes
-    VectorDims D_shape {N, DC}, S_shape {N, SC}, S_4D_shape {L, D, N, SC};
+    // Expected shapes.
+    const Shape shapeD{N, DC}, shapeS{N, SC};
 
-    if (in_data_dims != D_shape
-        || in_h_state_dims != S_shape
-        || out_h_state_dims != S_shape)
-        IE_THROW() << "Incorrect shape of input/output ports for layer " << getName();
+    if (getInputShapeAtPort(0).getInterval(1).isStatic() && getInputShapeAtPort(0) != shapeD ||
+            getInputShapeAtPort(1).getInterval(1).isStatic() && getInputShapeAtPort(1) != shapeS || getOutputShapeAtPort(0) != shapeS) {
+        THROW_ERROR << "has incorrect input/output shapes. Data shape: " << getInputShapeAtPort(0).toString() <<
+                "; Hidden state input: " << getInputShapeAtPort(1).toString() << "; Hidden state output: " << getOutputShapeAtPort(0).toString();
+    }
 
     if (S == 2) {
-        auto in_c_state_dims = op->get_input_shape(2);
-        auto out_c_state_dims = op->get_output_shape(1);
-
-        if (in_c_state_dims != S_shape
-            || out_c_state_dims != S_shape)
-            IE_THROW() << "Incorrect shape of input/output ports for layer " << getName();
+        if (getInputShapeAtPort(2).getInterval(1).isStatic() && getInputShapeAtPort(2) != shapeS || getOutputShapeAtPort(1) != shapeS)
+            THROW_ERROR << "has incorrect input/output shapes. Cell state input: " << getInputShapeAtPort(2).toString() <<
+                    "; Cell state output: " << getOutputShapeAtPort(1).toString();
     }
 }
 
 void MKLDNNRNN::fillCellDesc() {
-    runtimePrecision = getOriginalInputPrecisionAtPort(0);
-    auto dataType = MKLDNNExtensionUtils::IEPrecisionToDataType(runtimePrecision);
-
-    Shape S_4D_shape(VectorDims{L, D, N, SC});
+    const auto dataType = MKLDNNExtensionUtils::IEPrecisionToDataType(runtimePrecision);
+    const size_t B = N.getDummyValue(batchDimDummyValue);
```

Comment:
since we decided to remove `Interval` maybe better move this method into openvino/src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.h.
Declaration: `makeDummyDim(Dim minDim, Dim maxDim, Dim dummyVal)`
in case `minDim == maxDim return maxDim` in case `minDim != maxDim` do the same as https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp#L148
default `dummyVal == 64` the same as for `makeDummyDesc, makeDummyShape` 

after that we can reuse `makeDummyDim` here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/memory_desc/cpu_memory_desc_utils.cpp#L148

Reply:
makeDummyShape was reused in new approach without 'dummy dims'.